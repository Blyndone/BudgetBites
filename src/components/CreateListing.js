import * as React from "react";
import { View,  StyleSheet, SafeAreaView, Image } from 'react-native'
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Button, Text, TextInput, RadioButton
 } from 'react-native-paper';


const Separator = () => <View style={styles.separator} />;
const CreateListing = ({navigation}) => {

  var [name_text, setName] = React.useState("");
  var [desc_text, setDescription] = React.useState("");
  var [price_text, setPrice] = React.useState("");
  

  


    return (
        <SafeAreaView style={ styles.form}>
        
            
            <View style = { styles.textinput}>
            <Text style={styles.titleText}>
                Food Listing
            </Text>
            <Separator />
            <Text style = { styles.bodytext}>
                Create a new food listing. 
                {"\n"}
                {"\n"}
                 Input the item name, a short description, and the price below!
                {"\n"}
            </Text>




        <View 
        style= {{  
            padding: 10
        }}>
        
        
        <TextInput
      label="Item Name"
      value={name_text}
      onChangeText={name_text => setName(name_text)}
      style={styles.textinput}
      />


      
        <TextInput
      label="Description"
      value={desc_text}
      onChangeText={desc_text => setDescription(desc_text)}
      style={styles.textinput}

      />


      <TextInput
      label="Price"
      value={price_text}
      onChangeText={price_text => setPrice(price_text)}
      style={styles.textinput}
      keyboardType = 'number-pad'
      maxLength={10}
      />

      </View>



      </View>

            <View>

                    <Separator />
                    <Button mode = "contained"
                    title="List"
                    buttonColor="#eb6b34"
                    onPress={() => {
                      if( name_text.length ==0 || desc_text.length==0 || price_text.length ==0 ){
                        navigation.navigate('List')
                        return
                      }else{
                        try{
                          price_text = parseFloat(price_text).toFixed(2)
                        }catch (e){
                          price_text = "00.00"
                        }

                      }



                      fetch('http://10.0.2.2:5000/additem', {
                        method: 'POST',
                        headers: {
                          "Content-Type": "application/json"
                        },
                        body: JSON.stringify({
                          name_text: name_text,
                          desc_text: desc_text,
                          price_text: price_text
                        })
                      });
                      
                      navigation.navigate('List')
                      
                      
                    }
                    
                    }> Submit </Button>

                    <Separator />
                  

            </View>
        

        </SafeAreaView>
        
    )

}


const styles = StyleSheet.create({
    form: {
        flex: 1,
        backgroundColor: 'teal',
        justifyContent: 'top',
        padding: 20
    },
    titleText:{
        fontSize: 30,
        textAlign: 'left',
        color: 'black',
        fontWeight: 'bold',
        fontFamily: 'Helvetica'
    },
    button:{
        color:'#f194ff',
        backgroundColor: '#f194ff'
    },
    separator: {
        marginVertical: 8,
        borderBottomColor: '#737373',
        borderBottomWidth: StyleSheet.hairlineWidth,
      },
      textinput:{
        margin:5,


      },
      bodytext:{
        fontSize: 15,
        fontWeight: 'bold'
      },




})



export default CreateListing



