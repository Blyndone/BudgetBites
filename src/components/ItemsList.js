import React, { useEffect, useState } from 'react';
import { ActivityIndicator, FlatList, Text, View } from 'react-native';

const ItemsList = () => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState([]);

  const GetItems = async () => {
    try {
      const response = await fetch('http://10.0.2.2:5000/items');
      const json = await response.json();
      setData(json.items);
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    GetItems();
  }, []);

  return (
    <View>
      <Text>Items</Text>

      <View style={{ flex: 1, padding: 24 }}>
        {isLoading ? (
          <ActivityIndicator />
        ) : (
          <FlatList
            data={data}
            keyExtractor={({ itemID }) => itemID}
            renderItem={({ item }) => (
              <Text>
                {item.name}: {item.description}
              </Text>
            )}
          />
        )}
      </View>
    </View>
  );
};

export default ItemsList;
