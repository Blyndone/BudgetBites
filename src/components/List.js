import {
  Alert,
  Modal,
  View,
  Pressable,
  Text,
  StyleSheet,
  SafeAreaView,
  Image,
  Button,
  FlatList,
  StatusBar,
} from 'react-native';

import React, { useEffect, useState } from 'react';
import { Searchbar, Icon } from 'react-native-paper';
import images from '../../assets/testimages/ImageIndex.js';

const List = () => {
  const [modalVisible, setModalVisible] = useState(false);
  const [itemName, setItemName] = useState(0);
  const [itemDescripton, setItemDescription] = useState(0);
  const [itemImage, setItemImage] = useState(0);
  const [itemPrice, setItemPrice] = useState(0);
  const [itemID, setItemID] = useState(0);

  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState([]);

  const [searchQuery, setSearchQuery] = React.useState('');

  const GetItems = async () => {
    try {
      const response = await fetch('http://10.0.2.2:5000/getitems');
      const json = await response.json();
      console.log(searchQuery);
      console.log(searchQuery.length);

      if (!(searchQuery.length === 0)) {
        let results = Object.values(json.items).filter(
          (item) =>
            String(item.name)
              .toLowerCase()
              .includes(searchQuery.toLowerCase()) ||
            String(item.description)
              .toLowerCase()
              .includes(searchQuery.toLowerCase()),
        );
        setData(results);
      } else {
        setData(json.items);
      }
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    GetItems();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <Text style={styles.modalTitle}>{itemName}</Text>
            <View style={{ padding: 10 }}></View>
            <Image
              source={images[itemID]}
              style={{
                width: 150,
                height: 150,
              }}
            />
            <View style={{ padding: 10 }}></View>
            <Text style={styles.modalText}>{itemDescripton}</Text>
            <View style={{ padding: 10 }}></View>
            <Text style={styles.modalPrice}>${itemPrice}</Text>
            <View style={{ padding: 10 }}></View>
            <View style={{ flexDirection: 'row' }}>
              <Button
                mode="contained"
                title="Close"
                color="#eb6b34"
                onPress={() => setModalVisible(!modalVisible)}
              ></Button>
              <View style={{ padding: 10 }}></View>
              <Button
                mode="contained"
                title="Reserve"
                color="#eb6b34"
                onPress={() => setModalVisible(!modalVisible)}
              ></Button>
            </View>
          </View>
        </View>
      </Modal>

      <Text
        style={{
          fontWeight: 'bold',
          fontSize: 20,
          color: 'black',
        }}
      >
        {' '}
        List of Items
      </Text>

      <Searchbar
        placeholder="Search"
        onChangeText={setSearchQuery}
        value={searchQuery}
        onIconPress={GetItems}
        onSubmitEditing={GetItems}
        icon="camera"
      />
      <FlatList
        data={data}
        keyExtractor={({ itemID }) => itemID}
        renderItem={({ item }) => (
          <Pressable
            onPress={() => {
              setItemName(item.name);
              setItemImage(item.img);
              setItemID(item.itemID);
              setItemDescription(item.description);
              setItemPrice(item.price);
              setModalVisible(true);
            }}
          >
            <View style={styles.item}>
              <Image
                source={images[item.itemID]}
                style={{
                  width: 50,
                  height: 50,
                }}
              />

              <Text style={styles.instance}>{item.name}</Text>
              <Text style={styles.instance}>{item.description}</Text>
              <Text style={styles.instance}>{item.price}</Text>
            </View>
          </Pressable>
        )}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'teal',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: StatusBar.currentHeight || 0,
  },
  item: {
    flex: 1,
    padding: 5,
    marginVertical: 8,
    marginHorizontal: 16,
    height: 100,
    width: 360,
    flexDirection: 'row',
    // justifyContent: 'flex-end',
    alignItems: 'center',
    borderWidth: 5,
    backgroundColor: '#fca503',
  },
  titleText: {
    fontSize: 50,
    textAlign: 'center',
    color: 'black',
    fontWeight: 'bold',
    fontFamily: 'Helvetica',
  },
  button: {
    color: '#f194ff',
    backgroundColor: '#f194ff',
  },
  separator: {
    marginVertical: 8,
    borderBottomColor: '#737373',
    borderBottomWidth: StyleSheet.hairlineWidth,
  },
  instance: {
    textAlign: 'auto',
    flexBasis: 120,
    flexGrow: 1,
  },
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
  modalView: {
    margin: 20,
    backgroundColor: 'mediumturquoise',
    borderRadius: 20,
    padding: 20,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  modalTitle: {
    fontSize: 20,
    textAlign: 'center',
    color: 'black',
    fontWeight: 'bold',
    fontFamily: 'Helvetica',
  },
  modalText: {
    fontSize: 15,
    textAlign: 'center',
    color: 'black',
    fontWeight: 'normal',
    fontFamily: 'Helvetica',
  },
  modalPrice: {
    fontSize: 20,
    textAlign: 'center',
    color: 'black',
    fontWeight: 'bold',
    fontFamily: 'Helvetica',
  },
  button: {
    color: '#f194ff',
    backgroundColor: '#f194ff',
  },
});

export default List;
