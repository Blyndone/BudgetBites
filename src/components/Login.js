import React from "react";
import { View, Text, StyleSheet, SafeAreaView, Image, Button } from 'react-native'

const Separator = () => <View style={styles.separator} />;
const Login = () => {


    return (
        <SafeAreaView style={ styles.container}>
            <Image  
        source={require('../../assets/OIG1.png')}
        style={{
            width: 300, 
            height: 300,
        }}         
        />
        <Text style={styles.titleText}>
            Welcome to Budget Bites!
        </Text>
        <View>

        <Separator />
        <Button
        title="Login"
        color="#eb6b34"
      />
        <Separator />
        <Button
        title="Create Account"
        color="#eb6b34"
      />
      <Separator />
        <Button
        title="Guest Login"
        color="#eb6b34"
      />
        <Separator />
        </View>
        

        </SafeAreaView>
        
    )

}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'teal',
        justifyContent: 'center',
        alignItems: 'center'
    },
    titleText:{
        fontSize: 50,
        textAlign: 'center',
        color: 'black',
        fontWeight: 'bold',
        fontFamily: 'Helvetica'
    },
    button:{
        color:'#f194ff',
        backgroundColor: '#f194ff'
    },
    separator: {
        marginVertical: 8,
        borderBottomColor: '#737373',
        borderBottomWidth: StyleSheet.hairlineWidth,
      },



})



export default Login
