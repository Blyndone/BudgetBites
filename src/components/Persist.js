import * as SecureStore from 'expo-secure-store';
import { REACT_APP_ADDRESS } from '@env';
export async function Auth(user_text) {
  try {
    return await SecureStore.getItemAsync(user_text)
      .then((response) => {
        const myHeaders = new Headers();
        myHeaders.append('Content-Type', 'application/json');

        const raw = JSON.stringify({
          user_text: user_text,
          token: response,
        });

        const requestOptions = {
          method: 'POST',
          headers: myHeaders,
          body: raw,
          redirect: 'follow',
        };
        return requestOptions;
      })
      .then((requestOptions) => {
        return fetch(`${REACT_APP_ADDRESS}/auth`, requestOptions)
          .then((response) => response.text())
          .catch((error) => console.error(error));
      });
  } catch (err) {
    return err;
  }
}
export default Auth;
