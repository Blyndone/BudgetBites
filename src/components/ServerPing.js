
import React, {useEffect, useState} from 'react';
import {ActivityIndicator, FlatList, Text, View} from 'react-native';

const ServerPing = () => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState([]);

  const getMovies = async () => {
    try {
      const response = await fetch('http://10.0.2.2:5000/testtable');
      const json = await response.json();
      setData(json.testtable);
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => {
    getMovies();
  }, []);

  return (
    <View>
        <Text>Users</Text>


    <View style={{flex: 1, padding: 24}}>
      {isLoading ? (
          <ActivityIndicator />
          ) : (
              <FlatList
              data={data}
              keyExtractor={({id}) => id}
              renderItem={({item}) => (
                  <Text>
              {item.Name}, {item.Age}
            </Text>
          )}
          />
          )}
    </View>
          </View>
  );
};

export default ServerPing;