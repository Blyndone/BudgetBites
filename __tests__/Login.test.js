import { blah } from '../src/components/Login';

describe('Login component', () => {
  test('blah function adds two numbers correctly', () => {
    expect(blah(2, 3)).toBe(5);
  });
});
