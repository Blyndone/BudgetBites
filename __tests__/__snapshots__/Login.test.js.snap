// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "teal",
      "flex": 1,
      "justifyContent": "top",
      "padding": 20,
    }
  }
>
  <RCTScrollView
    keyboardShouldPersistTaps="never"
    scrollEnabled={false}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          {
            "alignContent": "center",
            "alignItems": "center",
            "justifyContent": "flex-start",
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 300,
              "width": 300,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    onLayout={[Function]}
    style={
      {
        "paddingBottom": 0,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#00b3b3",
          "borderRadius": 20,
          "marginVertical": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 10,
        }
      }
      testID="surface-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#00b3b3",
            "borderRadius": 20,
            "flex": undefined,
            "padding": 8,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 3,
          }
        }
        testID="surface"
      >
        <View
          style={
            {
              "margin": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "black",
                  "fontFamily": "Helvetica",
                  "fontSize": 30,
                  "fontWeight": "bold",
                  "textAlign": "left",
                },
              ]
            }
          >
            Login
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "fontSize": 15,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Enter details to login.
            

            Seller: User: seller | Pass: seller
            

            Buyer: User: customer | Pass: customer
          </Text>
          <View />
          <View
            style={
              {
                "padding": 10,
              }
            }
          >
            <View
              style={
                {
                  "margin": 5,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    false,
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                      "width": 750,
                      "zIndex": 3,
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "backgroundColor": "rgba(255, 251, 254, 1)",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 8,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "opacity": -0.6666666666666667,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 33,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                          {
                            "scaleY": 1,
                          },
                        ],
                        "width": -16,
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-background"
                  >
                    User Name
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    onLayout={[Function]}
                    onTextLayout={[Function]}
                    style={
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "maxWidth": 73,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-active"
                  >
                    User Name
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "maxWidth": 73,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-inactive"
                  >
                    User Name
                  </Text>
                </View>
                <TextInput
                  cursorColor="rgba(103, 80, 164, 1)"
                  editable={true}
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onLayout={[Function]}
                  placeholder=" "
                  placeholderTextColor="rgba(73, 69, 79, 1)"
                  selectionColor="rgba(103, 80, 164, 1)"
                  style={
                    [
                      {
                        "margin": 0,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "minWidth": 65,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                {
                  "margin": 5,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    false,
                    {
                      "backgroundColor": "rgba(255, 251, 254, 1)",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View
                style={
                  [
                    {
                      "paddingBottom": 0,
                    },
                    {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="none"
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": [
                        {
                          "translateX": 0,
                        },
                      ],
                      "width": 750,
                      "zIndex": 3,
                    }
                  }
                >
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "backgroundColor": "rgba(255, 251, 254, 1)",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 8,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "opacity": -0.6666666666666667,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 33,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                          {
                            "scaleY": 1,
                          },
                        ],
                        "width": -16,
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-background"
                  >
                    Password
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    onLayout={[Function]}
                    onTextLayout={[Function]}
                    style={
                      {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "maxWidth": 73,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-active"
                  >
                    Password
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "maxWidth": 73,
                        "opacity": 0,
                        "paddingHorizontal": 16,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 32,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateY": 0,
                          },
                          {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-inactive"
                  >
                    Password
                  </Text>
                </View>
                <TextInput
                  cursorColor="rgba(103, 80, 164, 1)"
                  editable={true}
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onLayout={[Function]}
                  placeholder=" "
                  placeholderTextColor="rgba(73, 69, 79, 1)"
                  secureTextEntry={true}
                  selectionColor="rgba(103, 80, 164, 1)"
                  style={
                    [
                      {
                        "margin": 0,
                      },
                      {
                        "height": 48,
                      },
                      {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "minWidth": 65,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      [
                        {},
                      ],
                      undefined,
                    ]
                  }
                  testID="text-input-outlined"
                  textContentType="password"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            style={
              {
                "margin": 10,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#eb6b34",
                  "borderRadius": 20,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container-outer-layer"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#eb6b34",
                    "borderColor": "transparent",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flex": undefined,
                    "minWidth": 64,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="button-container"
                title="Login"
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            [
                              {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              {
                                "marginHorizontal": 24,
                                "marginVertical": 10,
                              },
                              undefined,
                              false,
                              {
                                "color": "rgba(255, 255, 255, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "color": "black",
                                "fontSize": 18,
                              },
                            ],
                          ],
                        ]
                      }
                      testID="button-text"
                    >
                      Login
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "margin": 10,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#eb6b34",
                  "borderRadius": 20,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="button-container-outer-layer"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "#eb6b34",
                    "borderColor": "transparent",
                    "borderRadius": 20,
                    "borderStyle": "solid",
                    "borderWidth": 0,
                    "flex": undefined,
                    "minWidth": 64,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
                testID="button-container"
                title="CreateAccount"
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 20,
                      },
                    ]
                  }
                  testID="button"
                >
                  <View
                    style={
                      [
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "center",
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={1}
                      selectable={false}
                      style={
                        [
                          {
                            "textAlign": "left",
                          },
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "writingDirection": "ltr",
                          },
                          [
                            {
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            [
                              {
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              false,
                              {
                                "marginHorizontal": 24,
                                "marginVertical": 10,
                              },
                              undefined,
                              false,
                              {
                                "color": "rgba(255, 255, 255, 1)",
                                "fontFamily": "System",
                                "fontSize": 14,
                                "fontWeight": "500",
                                "letterSpacing": 0.1,
                                "lineHeight": 20,
                              },
                              {
                                "color": "black",
                                "fontSize": 16,
                              },
                            ],
                          ],
                        ]
                      }
                      testID="button-text"
                    >
                      Create Account
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
